content/articles/creating-an-animated-javafx-field-validation-alert/index.md

```diff
@@ -0,0 +1,758 @@
+- - `Fit Width`: 35
+- - `Fit Height`: 35
+- - `Layout X`: 40
+- - `Layout Y`: 400
+- **Code**
+- - `fx:id`: passwordIcon
+
+The SceneBuilder generated code is as follows:
+
+```xml
+      <ImageView fx:id="passwordIcon" fitHeight="35.0" fitWidth="35.0" layoutX="40.0" layoutY="400.0" pickOnBounds="true" preserveRatio="true">
+         <image>
+            <Image url="@lock.png" />
+         </image>
+      </ImageView>
+```
+
+- Add a Label control is required to display the errors generated or if successful.
+Add it by adding a Label control to the form. The label will not have any content on it.
+Rather it will have this content when the checks are done. The label will have the following:
+
+
+- **Properties**
+- - `Font`: System 12px
+- - `Text Fill`: RED
+- **Layout**
+- - `Pref Width`: 245
+- - `Layout X`: 80
+- - `Layout Y`: 445
+- **Code**
+- - `fx:id`: invalidDetails
+
+```xml
+      <!-- A label to display error and success messages -->
+      <Label fx:id="invalidDetails" layoutX="88.0" layoutY="446.0" prefWidth="245.0" textFill="RED" />
+```
+
+Buttons will be required to check if the inputs are entered as required or not, close the form, and reset the password.
+
+- Start by creating a **cancel button**. This closes the scene or window when clicked.
+Add a button from the controls. The button will have the following:
+
+
+- **Properties**
+- - `Text`: Cancel
+- - `Font`: System 14px
+- - `Text Fill`: #a3a3a3
+- **Style**
+- - `-fx-background-color`:  transparent
+- - `-fx-border-width`:  2
+- - `-fx-border-color`:  #a5a5a5
+- **Layout**
+- - `Pref Width`: 90
+- - `Pref Height`: 30
+- - `Layout X`: 85
+- - `Layout Y`: 480
+- **Code**
+- - `fx:id`: cancelButton
+- - `On Action`: onCancelButtonClick
+
+The `On Action` property identifies the button to a function. The button is the trigger.
+
+- Create a **login button** that has the following:
+
+
+- **Properties**
+- - `Text`: Login
+- - `Font`: System 15px
+- - `Text Fill`: #a3a3a3
+- **Style**
+- - `-fx-background-color`: #0077b6
+- - `-fx-border-radius`:  5
+- **Layout**
+- - `Pref Width`: 90
+- - `Pref Height`: 30
+- - `Layout X`: 225
+- - `Layout Y`: 480
+- **Code**
+- - `fx:id`: loginButton
+- - `On Action`: onLoginButtonClick
+
+Its code is as shown below:
+
+```xml
+      <Button fx:id="loginButton" layoutX="225.0" layoutY="480.0" mnemonicParsing="false" onAction="#onLoginButtonClick" prefHeight="30.0" prefWidth="90.0" style="-fx-background-color: #0077b6; -fx-border-radius: 5;" text="Login" textFill="WHITE">
+         <font>
+            <Font size="15.0" />
+         </font>
+      </Button>
+```
+
+- Add a **reset password button**. It has the following:
+
+
+- **Properties**
+- - `Text`: Login
+- - `Font`: System 12px
+- - `Text Fill`: #a3a3a3
+- **Style**
+- - `-fx-background-color`: transparent
+- - `-fx-border-radius`:  5
+- **Layout**
+- - `Pref Width`: 140
+- - `Pref Height`: 25
+- - `Layout X`: 120
+- - `Layout Y`: 530
+- **Code**
+- - `fx:id`: forgotButton

```

Comment:
If you already specified the values in the snippet there is no need of writing them separately. Do this for every instance you have a code snippet.